`cpal::SampleFormat` does not implement `Hash` or `Ord`

Without implementing [`Hash`] or [`Ord`], [`SampleFormat`] can't be used in some dynamic collections, particularly the sets from the standard library:

```rs
fn foo(formats: impl Iterator<Item = cpal::SampleFormat>) {
    use std::collections::{BTreeSet, HashSet};
    let _ = HashSet::from_iter(formats); //<<error[E0277]: the trait bound `SampleFormat: Hash` is not satisfied
    let _ = BTreeSet::from_iter(formats); //<<error[E0277]: the trait bound `SampleFormat: Ord` is not satisfied
}
```

I have a project where I'd like to let the user specify to my code certain requirements on the config of an input device, which involved constructing a set from a provided iterator of [`SampleFormat`]s. Without the type implementing [`Hash`] or [`Ord`] I'd need to use a [`Vec`] and manually de-duplicating formats from the input. Not the end of the world but it'd be nice to use a set instead.

[`Hash`]: https://doc.rust-lang.org/nightly/std/hash/trait.Hash.html
[`Ord`]: https://doc.rust-lang.org/nightly/std/cmp/trait.Ord.html
[`SampleFormat`]: https://docs.rs/cpal/0.15.3/cpal/enum.SampleFormat.html
[`Vec`]: https://doc.rust-lang.org/nightly/std/vec/struct.Vec.html

--- a/src/samples_formats.rs
+++ b/src/samples_formats.rs
@@ -6,7 +6,7 @@ pub use dasp_sample::{FromSample, Sample, I24, I48, U24, U48};
 
 /// Format that each sample has.
 #[cfg_attr(target_os = "emscripten", wasm_bindgen)]
-#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
 #[non_exhaustive]
 pub enum SampleFormat {
     /// `i8` with a valid range of 'u8::MIN..=u8::MAX' with `0` being the origin

